--- net/minecraft/server/packs/repository/PackCompatibility.java
+++ net/minecraft/server/packs/repository/PackCompatibility.java
@@ -1,7 +1,8 @@
 package net.minecraft.server.packs.repository;
 
+import net.minecraft.ChatFormatting;
 import net.minecraft.SharedConstants;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.TranslatableComponent;
 
 public enum PackCompatibility {
@@ -11,11 +12,11 @@
 
    private final Component description;
    private final Component confirmation;
 
    private PackCompatibility(String var3) {
-      this.description = new TranslatableComponent("pack.incompatible." + var3);
+      this.description = (new TranslatableComponent("pack.incompatible." + var3)).withStyle(ChatFormatting.GRAY);
       this.confirmation = new TranslatableComponent("pack.incompatible.confirm." + var3);
    }
 
    public boolean isCompatible() {
       return this == COMPATIBLE;
