--- net/minecraft/server/packs/repository/PackSource.java
+++ net/minecraft/server/packs/repository/PackSource.java
@@ -1,7 +1,8 @@
 package net.minecraft.server.packs.repository;
 
+import net.minecraft.ChatFormatting;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.TranslatableComponent;
 
 public interface PackSource {
    PackSource DEFAULT = passThrough();
@@ -18,9 +19,9 @@
    }
 
    static PackSource decorating(String var0) {
       TranslatableComponent var1 = new TranslatableComponent(var0);
       return (var1x) -> {
-         return new TranslatableComponent("pack.nameAndSource", new Object[]{var1x, var1});
+         return (new TranslatableComponent("pack.nameAndSource", new Object[]{var1x, var1})).withStyle(ChatFormatting.GRAY);
       };
    }
 }
