--- net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -51,16 +51,16 @@
       DataInputStream var3 = var2.getChunkDataInputStream(var1);
       Throwable var4 = null;
 
       CompoundTag var5;
       try {
-         if (var3 != null) {
-            var5 = NbtIo.read((DataInput)var3);
+         if (var3 == null) {
+            var5 = null;
             return var5;
          }
 
-         var5 = null;
+         var5 = NbtIo.read((DataInput)var3);
       } catch (Throwable var15) {
          var4 = var15;
          throw var15;
       } finally {
          if (var3 != null) {
