--- com/mojang/realmsclient/util/RealmsUtil.java
+++ com/mojang/realmsclient/util/RealmsUtil.java
@@ -10,16 +10,15 @@
 import com.mojang.authlib.minecraft.MinecraftProfileTexture.Type;
 import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
 import com.mojang.util.UUIDTypeAdapter;
 import java.util.Date;
 import java.util.Map;
-import java.util.UUID;
 import java.util.concurrent.TimeUnit;
 import net.minecraft.client.Minecraft;
 
 public class RealmsUtil {
-   private static final YggdrasilAuthenticationService AUTHENTICATION_SERVICE = new YggdrasilAuthenticationService(Minecraft.getInstance().getProxy(), UUID.randomUUID().toString());
+   private static final YggdrasilAuthenticationService AUTHENTICATION_SERVICE = new YggdrasilAuthenticationService(Minecraft.getInstance().getProxy());
    private static final MinecraftSessionService SESSION_SERVICE;
    public static LoadingCache<String, GameProfile> gameProfileCache;
 
    public static String uuidToName(String var0) throws Exception {
       GameProfile var1 = (GameProfile)gameProfileCache.get(var0);
