--- com/mojang/authlib/yggdrasil/request/ValidateRequest.java
+++ com/mojang/authlib/yggdrasil/request/ValidateRequest.java
@@ -1,13 +1,11 @@
 package com.mojang.authlib.yggdrasil.request;
 
-import com.mojang.authlib.yggdrasil.YggdrasilUserAuthentication;
-
 public class ValidateRequest {
    private String clientToken;
    private String accessToken;
 
-   public ValidateRequest(YggdrasilUserAuthentication var1) {
-      this.clientToken = var1.getAuthenticationService().getClientToken();
-      this.accessToken = var1.getAuthenticatedToken();
+   public ValidateRequest(String var1, String var2) {
+      this.clientToken = var2;
+      this.accessToken = var1;
    }
 }
