--- net/minecraft/client/gui/components/toasts/TutorialToast.java
+++ net/minecraft/client/gui/components/toasts/TutorialToast.java
@@ -66,11 +66,12 @@
    public static enum Icons {
       MOVEMENT_KEYS(0, 0),
       MOUSE(1, 0),
       TREE(2, 0),
       RECIPE_BOOK(0, 1),
-      WOODEN_PLANKS(1, 1);
+      WOODEN_PLANKS(1, 1),
+      SOCIAL_INTERACTIONS(2, 1);
 
       private final int x;
       private final int y;
 
       private Icons(int var3, int var4) {
