--- net/minecraft/world/level/material/FlowingFluid.java
+++ net/minecraft/world/level/material/FlowingFluid.java
@@ -190,11 +190,15 @@
       FluidState var15 = var14.getFluidState();
       if (!var15.isEmpty() && var15.getType().isSame(this) && this.canPassThroughWall(Direction.UP, var1, var2, var3, var12, var14)) {
          return this.getFlowing(8, true);
       } else {
          int var16 = var4 - this.getDropOff(var1);
-         return var16 <= 0 ? Fluids.EMPTY.defaultFluidState() : this.getFlowing(var16, false);
+         if (var16 <= 0) {
+            return Fluids.EMPTY.defaultFluidState();
+         } else {
+            return this.getFlowing(var16, false);
+         }
       }
    }
 
    private boolean canPassThroughWall(Direction var1, BlockGetter var2, BlockPos var3, BlockState var4, BlockPos var5, BlockState var6) {
       Object2ByteLinkedOpenHashMap var7;
