--- net/minecraft/DetectedVersion.java
+++ net/minecraft/DetectedVersion.java
@@ -26,17 +26,17 @@
    private final Date buildTime;
    private final String releaseTarget;
 
    private DetectedVersion() {
       this.id = UUID.randomUUID().toString().replaceAll("-", "");
-      this.name = "1.16.3";
+      this.name = "1.16.4";
       this.stable = true;
-      this.worldVersion = 2580;
-      this.protocolVersion = 753;
+      this.worldVersion = 2584;
+      this.protocolVersion = SharedConstants.getProtocolVersion();
       this.packVersion = 6;
       this.buildTime = new Date();
-      this.releaseTarget = "1.16.3";
+      this.releaseTarget = "1.16.4";
    }
 
    private DetectedVersion(JsonObject var1) {
       this.id = GsonHelper.getAsString(var1, "id");
       this.name = GsonHelper.getAsString(var1, "name");
