--- net/minecraft/client/gui/screens/SimpleOptionsSubScreen.java
+++ net/minecraft/client/gui/screens/SimpleOptionsSubScreen.java
@@ -25,20 +25,24 @@
 
    protected void init() {
       this.list = new OptionsList(this.minecraft, this.width, this.height, 32, this.height - 32, 25);
       this.list.addSmall(this.smallOptions);
       this.children.add(this.list);
-      this.addButton(new Button(this.width / 2 - 100, this.height - 27, 200, 20, CommonComponents.GUI_DONE, (var1) -> {
-         this.minecraft.setScreen(this.lastScreen);
-      }));
+      this.createFooter();
       this.narratorButton = this.list.findOption(Option.NARRATOR);
       if (this.narratorButton != null) {
          this.narratorButton.active = NarratorChatListener.INSTANCE.isActive();
       }
 
    }
 
+   protected void createFooter() {
+      this.addButton(new Button(this.width / 2 - 100, this.height - 27, 200, 20, CommonComponents.GUI_DONE, (var1) -> {
+         this.minecraft.setScreen(this.lastScreen);
+      }));
+   }
+
    public void render(PoseStack var1, int var2, int var3, float var4) {
       this.renderBackground(var1);
       this.list.render(var1, var2, var3, var4);
       drawCenteredString(var1, this.font, this.title, this.width / 2, 20, 16777215);
       super.render(var1, var2, var3, var4);
